File: java/Singleton.java
import java.util.HashMap;
import java.util.Map;

public class Singleton {
    private static Singleton instance;
    private Map<String, String> parametres;

    private Singleton() {
        parametres = new HashMap<>();
    }

    public static Singleton getInstance() {
        if (instance == null) {
            instance = new Singleton();
        }
        return instance;
    }

    public void definir(String cle, String valeur) {
        parametres.put(cle, valeur);
    }

    public String obtenir(String cle) {
        return parametres.get(cle);
    }

    public static void main(String[] args) {
        Singleton config1 = Singleton.getInstance();
        config1.definir("langue", "fr");

        Singleton config2 = Singleton.getInstance();
        System.out.println(config2.obtenir("langue"));
        System.out.println("Même instance : " + (config1 == config2));
    }
}

File: python/singleton.py
class ConfigManager:
    _instance = None

    def __new__(cls):
        if cls._instance is None:
            cls._instance = super().__new__(cls)
            cls._instance.parametres = {}
        return cls._instance

    def definir(self, cle, valeur):
        self.parametres[cle] = valeur

    def obtenir(self, cle):
        return self.parametres.get(cle)


if __name__ == "__main__":
    config1 = ConfigManager()
    config1.definir("langue", "fr")

    config2 = ConfigManager()
    print(config2.obtenir("langue"))
    print("Même instance :", config1 is config2)
